data/pigui/modules/new-game-window/location.lua

```diff
@@ -277,13 +289,84 @@ function Location:fromStartVariant(variant)
 		path = variant.location,
 		state = State.UNKNOWN
 	}
-	self:setLock(true)
+	self.lock = true
 end
 
 function Location:isValid()
 	return self.bodySelected and true or false
 end
 
+function Location:updateParams()
+	self:setPath(self.value.path)
+end
+
+---@param id number
+---@param frame table
+---@return number?
+local function getSystemBodyIndexForFrameID(id, frame)
+	if id == frame.frameId then
+		return frame.index_for_system_body
+	end
+	if not frame.child_frames then
+		return nil
+	end
+	for _, childFrame in pairs(frame.child_frames) do
+		local sbi = getSystemBodyIndexForFrameID(id, childFrame)
+		if sbi then return sbi end
+	end
+end
+
+Location.reader = Helpers.versioned {{
+	version = 89,
+	fnc = function(saveGame)
+
+		local system, errorString = Helpers.getByPath(saveGame, "space/star_system")
+		if errorString then return nil, errorString end
+
+		local path = SystemPath.New(system.sector_x, system.sector_y, system.sector_z, system.system_index)
+		local sector = Location:getGalaxy():GetSector(path.sectorX, path.sectorY, path.sectorZ)
+		if path.systemIndex >= #sector then
+			return nil, "Bad system index"
+		end
+
+		-- the indices of the bodies and the corresponding system bodies
+		-- (Space::m_bodyIndex and Space::m_sbodyIndex) turn out to be the same
+		local dockedTo
+		dockedTo, errorString = Helpers.getPlayerShipParameter(saveGame, "ship/index_for_body_docked_with")
+		if errorString then return nil, errorString end
+
+		if dockedTo ~= 0 then
+			-- bodyIndex in system path is 1 less than index in Space::m_sbodyIndex,
+			dockedTo = dockedTo - 1
+			local starSystem = Location:getGalaxy():GetStarSystem(path)
+			local paths = starSystem:GetBodyPaths()
```

Comment:
Unless you're using this `paths` variable somewhere else, I'd recommend using the attr `starSystem.numberOfBodies` instead - it doesn't push new SystemPath objects to Lua for every (potentially 70+) body in the system.

Reply:
Done!